### DISCUSSION TITLE
Why doesn't Fabric pipe commands correctly?

### DISCUSSION THREAD

**Discussion by x-Jack-8276** [danielmiessler/Fabric#825] [Q&A]
Fabric used to work exactly like the video demonstrated (even before the go update),  however for some reason it stopped working. 

Any help is gratefully appreciated.

eg: ```yt https://youtu.be/wPEyyigh10g | fabric -p rate_content ```

output:
```**Summary of Key Points:**

1. **AI Enhancement, Not Replacement:** The speaker emphasizes that they're using AI to enhance human activities, not replace them. They give an example of a group playing a tabletop game with miniatures every Friday, and how AI can enhance this experience without replacing it.

2. **AI Extracts Meaningful Events and Dialogue:** The AI records key events and interesting dialogue from the gaming sessions, creating a memorable summary that can be shared among players.

3. **AI Generates Dramatic Narration:** An AI voice (from 11 Labs) reads out a setup for the next episode in the style of old TV shows, adding a fun, dramatic element to the game.

4. **Human-First Approach to AI:** The speaker's approach to AI is to solve human problems and enhance human activities with technology and AI, rather than focusing on replacement.

5. **Fabric Project Installation:**
   - Go to the quick start link.
   - Download the installer (repo).
   - Install Poetry using a curl command.
   - Make the setup file executable and run it.
   - Run `fabric setup` to install your key from OpenAI.
   - Verify installation by running `fabric help`.

6. **Encouragement for Human-First AI:** The speaker encourages users to adopt a human-first approach with AI, viewing it as an augmentation rather than replacement.

7. **Future Plans:**
   - More videos coming out soon showcasing use cases and patterns.
   - Looking forward to seeing what users do with the project.
```

**However if I bypass fabric but still use the patterns it works correctly:**

eg:  ```yt https://youtu.be/wPEyyigh10g | ollama run $OLLAMA_MODEL $(cat ~/fabric/patterns/rate_content/syst
em.md) ```

output:
```
**LABELS:** AI, Technology, Humanity, Enhancement, Philosophy

**RATING:** B Tier: (Consume Original When Time Allows)

Explanation:
- 15+ ideas discussed.
- Decent theme matching with "human meaning", "the future of AI", and "abstract thinking".
- Some themes not strongly addressed include "mental models" and "meaning in a post-AI world".

**CONTENT SCORE:** 78/100

Explanation:
- Many ideas were presented, but some could have been developed further.
- The philosophy behind AI augmentation was well-articulated, but specific examples of human-first AI applications
could have been provided for better illustration.
- The installation guide for Fabric was thorough but could have been more concise to maintain viewer engagement.
- The speaker's passion and enthusiasm were infectious, contributing to the overall quality of the content.
```

**Comment by nic0711** [comment]
Same here https://github.com/danielmiessler/fabric/discussions/841
I tried the bypass and it worked fine... (with extract_wisdom_large)

**Reply by x-Jack-8276** [reply]
this post here [568](https://github.com/danielmiessler/fabric/discussions/568) seems to be the same issue and a solution is provided here [568-solution](https://github.com/danielmiessler/fabric/discussions/568#discussioncomment-10287854)
I have not tested the solution

**Comment by schmidthaus** [comment]
> Fabric used to work exactly like the video demonstrated (even before the go update), however for some reason it stopped working.
> 
> Any help is gratefully appreciated.
> 
> eg: `yt https://youtu.be/wPEyyigh10g | fabric -p rate_content `
> 
> output:

No piping works on my Mac after the latest update (Go version) using:
`go install -ldflags "-X main.version=$(git describe --tags --always)" github.com/danielmiessler/fabric@latest`
`go: downloading golang.org/x/text v0.18.0
go: downloading github.com/sashabaranov/go-openai v1.30.0
go: downloading golang.org/x/sys v0.25.0
go: downloading github.com/liushuangls/go-anthropic/v2 v2.8.0
go: downloading github.com/google/generative-ai-go v0.18.0
go: downloading google.golang.org/api v0.197.0
go: downloading github.com/ollama/ollama v0.3.11
go: downloading dario.cat/mergo v1.0.1
go: downloading golang.org/x/net v0.29.0
go: downloading golang.org/x/crypto v0.27.0
go: downloading github.com/cyphar/filepath-securejoin v0.3.2
go: downloading cloud.google.com/go/ai v0.8.2
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20240903143218-8af14fe29dc1
go: downloading google.golang.org/grpc v1.66.2
go: downloading cloud.google.com/go v0.115.1
go: downloading github.com/cloudflare/circl v1.4.0
go: downloading github.com/skeema/knownhosts v1.3.0
go: downloading cloud.google.com/go/longrunning v0.6.1
go: downloading google.golang.org/genproto/googleapis/api v0.0.0-20240903143218-8af14fe29dc1
go: downloading cloud.google.com/go/auth v0.9.4
go: downloading golang.org/x/oauth2 v0.23.0
go: downloading cloud.google.com/go/auth/oauth2adapt v0.2.4
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.55.0
go: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.55.0
go: downloading cloud.google.com/go/compute/metadata v0.5.1
go: downloading go.opentelemetry.io/otel v1.30.0
go: downloading go.opentelemetry.io/otel/metric v1.30.0
go: downloading go.opentelemetry.io/otel/trace v1.30.0
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.4`

I've tried piping the following approaches thus far with no joy:
`cat result.txt | fabric --pattern explain_terms`
OUTPUT: Please provide the content you would like me to create a glossary for.

`pbpaste | fabric --pattern explain_terms` (I did confirm the pbpaste content first)
OUTPUT: Please provide the content you would like me to create a glossary for.

It worked in the previous version of fabric (Go).

Any ideas on how to resolve?

**Comment by BryanKerrEdTech** [comment]
I'm also seeing apparent issues with piping.

`pbpaste | fabric -p extract_wisdom > fout.txt`

Gives me:
`I apologize, but I don't have enough context or content to provide a full analysis based on the instructions given. The input appears to be a brief greeting "Hi" which doesn't contain enough information to extract insights, quotes, facts, etc. If you'd like me to analyze a specific piece of content, please provide the full text or transcript and I'd be happy to apply the requested analysis steps.`

But
`pbpaste | fabric -p rate_content > fout.txt`
and 
`pbpaste | fabric -p summarize > fout.txt`
work fine.
